url354ce1a2d1_dagaren_gladiator_chess_tgz-pJ8-es_dagaren_gladiator_main_MainJ8/src/es/dagaren/gladiator/search/AlphaBetaSearcher.java

```diff
@@ -0,0 +1,625 @@
+         }
+         logger.info("Variante principal en la profundidad " + depth + ": " + pv);
+         logger.info("Mejor puntuación en la profundidad " + depth + ": " + bestScore);
+         
+         
+         int nodesPercent = (nodes + qnodes) > 0 ? ((100 * nodes) / (nodes + qnodes)) : 0;
+         int qnodesPercent = (nodes + qnodes) > 0 ? ((100 * qnodes) / (nodes + qnodes)) : 0;
+         int cutoffsPercent = (cutoffs + nodes) > 0 ? ((100 * cutoffs) / (cutoffs + nodes)) : 0;
+         int qcutoffsPercent = (qcutoffs + qnodes) > 0 ? ((100 * qcutoffs) / (qcutoffs + qnodes)) : 0;
+         int aspirationFailsPercent = aspirationSearchs > 0 ? (100 * aspirationSearchsFails) / aspirationSearchs : 0;
+         int pvsHitsPercent = (pvsHits + pvsFails) > 0 ? (100 * pvsHits) / (pvsHits + pvsFails) : 0;
+         int withTranspositionPercent = (withTranspositionMove + withoutTranspositionMove) > 0 ? (100 * withTranspositionMove) / (withTranspositionMove + withoutTranspositionMove) : 0;
+         
+         logger.debug("Estadísticas búsquedas en profundidad " + depth + ":");
+         logger.debug("------------------------------------------");
+         logger.debug(" * Nodos recorridos: " + nodes + "(" + nodesPercent + "%)");
+         logger.debug(" * Nodos quiescende recorridos: " + qnodes + "(" + qnodesPercent + "%)");
+         logger.debug(" * Cortes producidos en nodos normales: " + cutoffs + "(" + cutoffsPercent  +"%)");
+         logger.debug(" * Cortes producidos en nodos quiescence: " + qcutoffs + "(" + qcutoffsPercent  +"%)");
+         logger.debug(" * Fallos en ventana de aspiración: " + aspirationSearchsFails + "(" + aspirationFailsPercent  +"%)");
+         logger.debug(" * Fallos en ventana de aspiración por inestabilidad: " + aspirationSearchInstabilityFails);
+         logger.debug(" * Tabla de tranposición: Aciertos: " + transpositionTable.getHits() + ", Fallos: " + transpositionTable.getMisses());
+         logger.debug(" * Cortes por tabla de transposición: " + transpositionCutoffs);
+         logger.debug(" * Aciertos PVS: " + pvsHits + "(" + pvsHitsPercent + "%)");
+         logger.debug(" * Fallos   PVS: " + pvsFails + "(" + (100 - pvsHitsPercent) + "%)");
+         logger.debug(" * Busquedas con transposicion: " + withTranspositionMove + "(" + withTranspositionPercent + "%)");
+         logger.debug(" * Busquedas sin transposicion: " + withoutTranspositionMove + "(" + (100 - withTranspositionPercent) + "%)");
+         logger.debug(" * Futility prunings: " + futilityPrunings);
+         
+         
+         logger.debug(" * Tiempo iteración: " + ((System.currentTimeMillis() - iterationInitTime) / 1000));
+         
+         //Si el valor es mate se termina la búsqueda
+         if(score == CHECKMATE_SCORE || score == -CHECKMATE_SCORE)
+         {
+            break;
+         }  
+      }
+      
+   }
+   
+   public int alphaBeta(Position position, int alpha, int beta, int currentDepth, int remainingPlies)
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 24. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test.

We recommend that you simplify this method or break it into multiple methods. For example, consider extracting the code block on lines 259-451 into a separate method.